File: backend/app/domain/repositories/play_repository.py
from abc import ABC, abstractmethod
from datetime import date
from uuid import UUID

from app.domain.entities.play_record import PlayRecord


class PlayRepository(ABC):
    @abstractmethod
    async def get(self, id: UUID) -> PlayRecord | None:
        ...

    @abstractmethod
    async def get_by_baby_and_date(self, baby_id: UUID, target_date: date) -> list[PlayRecord]:
        ...

    @abstractmethod
    async def save(self, record: PlayRecord) -> PlayRecord:
        ...

    @abstractmethod
    async def update(self, record: PlayRecord) -> PlayRecord:
        ...

    @abstractmethod
    async def delete(self, id: UUID) -> None:
        ...

File: zzippu_ios/zzippu/Domain/Repositories/PlayRepository.swift
// Domain/Repositories/PlayRepository.swift
// Foundation only — SwiftUI/SwiftData import 금지

import Foundation

protocol PlayRepository {
    func create(_ play: PlayRecord) async throws -> PlayRecord
    func delete(id: UUID, babyId: UUID) async throws
    func list(babyId: UUID, on day: Date) async throws -> [PlayRecord]
}
